bundles/core/org.eclipse.smarthome.core.audio/src/main/java/org/eclipse/smarthome/core/audio/URLAudioStream.java

```diff
@@ -52,27 +57,34 @@ public URLAudioStream(String url) throws AudioException {
 
     private InputStream createInputStream() throws AudioException {
         try {
-            if (url.toLowerCase().endsWith(".m3u")) {
-                InputStream is = new URL(url).openStream();
-                String urls = IOUtils.toString(is);
-                for (String line : urls.split("\n")) {
-                    if (!line.isEmpty() && !line.startsWith("#")) {
-                        url = line;
-                        break;
-                    }
-                }
-            } else if (url.toLowerCase().endsWith(".pls")) {
-                InputStream is = new URL(url).openStream();
-                String urls = IOUtils.toString(is);
-                for (String line : urls.split("\n")) {
-                    if (!line.isEmpty() && line.startsWith("File")) {
-                        Matcher matcher = plsStreamPattern.matcher(line);
-                        if (matcher.find()) {
-                            url = matcher.group(1);
+            final String filename = url.toLowerCase();
+            final String extension = AudioStreamUtils.getExtension(filename);
+            switch (extension) {
+                case M3U_EXTENSION:
+                    InputStream isM3U = new URL(url).openStream();
+                    String urlsM3U = IOUtils.toString(isM3U);
+                    for (String line : urlsM3U.split("\n")) {
```

Comment:
Line 65 & 66 might also be a call to `IOUtils.readLines(is)` which does provide a List\<String\> ready to use.